VerilogForVerificationStyle.md

```diff
@@ -0,0 +1,826 @@
+  * [Sequence Items](#sequence-items)
+  * [Objections and Coordinating End of Test](#objections-and-coordinating-end-of-test)
+  * [Logging and Print Messages](#logging-and-print-messages)
+  * [DPI and C Connections](#dpi-and-c-connections)
+* [SystemVerilog Language Features](#systemverilog-language-features)
+  * [Randomization](#randomization)
+  * [Enums](#enums)
+  * [Interfaces, Clocking Blocks, Modports](#interfaces,-clocking-blocks,-modports)
+  * [Loop Operators](#loop-operators)
+  * [Code Within Asserts](#code-within-asserts)
+  * [Wait and Fork](#wait-and-fork)
+  * [Void Casts](#void-casts)
+  * [Associative Arrays](#associative-arrays)
+  * [Bind Statements](#bind-statements)
+  * [Simulator Specific Code](#simulator-specific-code)
+  * [Forbidden System Tasks and Functions](#forbidden-system-tasks-and-functions)
+
+
+
+## Introduction
+
+This guide defines the Comportable style for DV-specific SystemVerilog code. The
+goal is to use this common set of guidelines when implementing testbenches using
+the UVM methodology to allow all developed Comportable DV code to be uniform,
+reusable, and portable.
+This document assumes previous experience and working knowledge of the UVM, and
+as such all included code snippets will be fairly concise.
+A testbench generation script
+[uvmdvgen.py](https://github.com/lowRISC/opentitan/blob/master/util/uvmdvgen.py)
+is used to automatically generate boilerplate DV testbench classes that follow
+these style guidelines.
+
+## Naming and Style
+
+Refer to [lowRISC Verilog naming
+conventions](https://github.com/lowRISC/style-guides/blob/master/VerilogCodingStyle.md#naming)
+for any general naming style guidelines. In general, common UVM testbench
+components will be declared and assigned handles as follows:
+
+| Testbench component               | Style                                        |
+| --------------------------------- | -------------------------------------------- |
+| Virtual interfaces                | `virtual <if_name>_if <if_name>_vif;`        |
+| Environment                       | `<dut>_env env;`                             |
+| Env level config object           | `rand <dut>_env_cfg cfg;`                    |
+| Env coverage collection object    | `<dut>_env_cov cov;`                         |
+| Agent                             | `<dut>_agent m_<dut>_agent;`                 |
+| Agent level config object         | `rand <dut>_agent_cfg m_<dut>_agent_cfg;`    |
+| Agent coverage collection object  | `<dut>_agent_cov cov;`                       |
+| Driver                            | `<dut>_driver driver;`                       |
+| Monitor                           | `<dut>_monitor monitor;`                     |
+| Scoreboard                        | `<dut>_scoreboard scoreboard;`               |
+| Virtual sequencer                 | `<dut>_virtual_sequencer virtual_sequencer;` |
+| Sequencer                         | `<dut>_sequencer <dut>_sequencer;`           |
```

Comment:
Additional naming guidelines:
Testbench file that instantiates the DUT: `tb.sv`
Testbench module name: `tb`
Instance name of the DUT in `tb.sv`: `dut`

Standardizing these naming conventions makes it useful to create supporting DV sources that work universally across all IPs (example: `hw/dv/tools/vcs/cover.cfg` which tells VCS what coverage patterns to collect). Without the standard tb and dut names, each IP would have to create its own version which slightly increases effort.

Reply:
Updated, thanks for the suggestion!